pkg/resources/connection.go

```diff
@@ -0,0 +1,321 @@
+package resources
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/helpers"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/provider"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/schemas"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var connectionSchema = map[string]*schema.Schema{
+	"name": {
+		Type:             schema.TypeString,
+		Required:         true,
+		ForceNew:         true,
+		Description:      blocklistedCharactersFieldDescription("String that specifies the identifier (i.e. name) for the connection. Must start with an alphabetic character and may only contain letters, decimal digits (0-9), and underscores (_). For a primary connection, the name must be unique across connection names and account names in the organization. For a secondary connection, the name must match the name of its primary connection."),
+		DiffSuppressFunc: suppressIdentifierQuoting,
+	},
+	"as_replica_of": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		ForceNew:         true,
+		Description:      "Specifies the identifier for a primary connection from which to create a replica (i.e. a secondary connection).",
+		DiffSuppressFunc: suppressIdentifierQuoting,
+	},
```

Comment:
Do we want to have ForceNew here?

Reply:
As far as i understand the documentation correctly, replicated connection can be promoted to serve as primary connection, which essentially (to me) means that it can create a chain of replications

I agree that the best solution would be to somehow transfer secondary to primary connection, but how can this be achieved? Thats why i've decided to leave it as one resource, since secondary connection can be promoted to and serve as primary

There is no good solution for now, since connection as one resource has more responsibility and its state (primary or secondary) is dependent on `is_primary`. On the other hand how can we transfer secondary_connection resource to primary one in a way that won't be confusing to users?

Reply:
I have answered to the thread internally; IMO until we clarify the behavior on SF side, let's keep to separate resources and the promotion would mean dropping and recreating (not by forceNew but removal of old resource in config and adding the new one).

Of course, we have to also then react to external type changes (after split).

Reply:
What if the promotion of secondary connection to primary has been made externally? How should we react? as i understand there is no way to demote the primary back to secondary unless its "parent" removes the account that created secondary connection for `enable failover to accounts` list 

One option i can think of is that when we detect external change of `show_output.is_primary` from false to true, then we alter the parent of secondary connection and remove the account which created it from the list

Reply:
We should force recreation if there is no suitable alter (maybe let's handle external promotion in a separate PR?)

Reply:
I think this can be resolved.